includes/admin/MessagesRepository.php

```diff
@@ -21,30 +25,68 @@ class MessagesRepository
     public function __construct()
     {
         add_action('admin_notices', [$this, 'showMessages']);
+        add_action('wp_ajax_dismissNotice', [$this, 'dismissNotice']);
     }
 
     /**
-     * @param  string $message
-     * @param  string $level
-     * @param  bool   $onAllPages
+     * @param  string      $message
+     * @param  string      $level
+     * @param  string|null $messageId
+     * @param  array       $onPages
      */
-    public function addMessage(string $message, string $level, bool $onAllPages = false): void
+    public function addMessage(string $message, string $level, ?string $messageId, array $onPages = []): void
     {
         $this->messages[] = [
-            'message'    => $message,
-            'level'      => $level,
-            'onAllPages' => $onAllPages,
+            'message'   => $message,
+            'messageId' => $messageId ?? null,
+            'level'     => $level,
+            'onPages'   => $onPages,
         ];
     }
 
     public function showMessages(): void
     {
         foreach ($this->messages as $message) {
-            if (! $message['onAllPages'] && ! WCMYPA_Settings::isViewingOwnSettingsPage()) {
-                continue;
+            if ($this->shouldMessageBeShown($message)) {
+                $isDismissible = $message['messageId'] ? 'is-dismissible' : '';
+                echo sprintf(
```

Comment:
`echo sprintf` -> `printf`

De static analysis tool wordt helemaal gek hier, misschien kan je kijken naar hoe je dit moet escapen in de WordPress Developer Handbooks die vermeld wordt